extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ObjectQueryService.java

```diff
@@ -141,16 +141,21 @@ public ExprNode generateQuery(ExprNode filter,
         // the given search filter
         AndNode searchFilter = new AndNode();
         searchFilter.addNode(filter);
-
-        // Include all attributes within OR clause if there are more than one
-        OrNode attributeFilter = new OrNode();
+        
+        // Add in attribute list
+        if (attributes.size() > 0) {
```

Comment:
If we should be adding nothing in the case where there are no attributes, would it make more sense to simply `return filter` if no attributes are given?

Reply:
Yep, that's fair - implemented as suggested.